src/views/SessionView/index.vue

```diff
@@ -168,16 +180,14 @@ export default {
       });
     },
     reconnect_attempt() {
-      this.sessionReconnecting = true;
+      this.$store.dispatch("user/sessionDisconnected");
+      if (!this.session || !this.session._id) {
+        this.showTroubleStartingModal();
+      }
     },
     connect() {
-      if (this.sessionReconnecting) {
-        if (this.session && this.session._id) {
-          // we still need to re-join the room after Socket.IO re-establishes the connection
-          this.joinSession(this.session._id);
-        } else {
-          location.reload();
-        }
+      if (!this.isSessionConnectionAlive) {
+        this.joinSession(this.session._id);
```

Comment:
are we sure `this.session` won't be undefined? seems like thats something thats being checked in `reconnect_attempt` but not here. if you want to avoid dealing with this, lodash's `get` will let you grab a value inside an object and set a default value if the key (or object) doesnt exist: [lodash get](https://lodash.com/docs/4.17.15#get)

Reply:
This is a good point. I've edited this code to check whether this.session is defined first.